bindings/cpp/iiopp.h

```diff
@@ -309,221 +389,193 @@ class Channel
-    void enable() {iio_channel_enable(p);}
-    optstr find_attr(cstr name) {return impl::opt(iio_channel_find_attr(p, name));}
-    unsigned int attrs_count() const {return iio_channel_get_attrs_count(p);}
-    void * data() const {return iio_channel_get_data(p);}
-    Device device();
+    Device device() const;
     cstr id() const {return iio_channel_get_id(p);}
-    iio_modifier modifier() const {return iio_channel_get_modifier(p);}
     optstr name() const { return impl::opt(iio_channel_get_name(p));}
-    iio_chan_type type() const {return iio_channel_get_type(p);}
-    bool is_enabled() const { return iio_channel_is_enabled(p);}
     bool is_output() const { return iio_channel_is_output(p);}
     bool is_scan_element() const { return iio_channel_is_scan_element(p);}
-    size_t read(Buffer buffer, void * dst, size_t len) const; // Flawfinder: ignore
-    size_t read_raw(Buffer buffer, void * dst, size_t len) const;
+    unsigned int attrs_count() const {return iio_channel_get_attrs_count(p);}
+    Attr attr(unsigned int index) {return iio_channel_get_attr(p, index);}
+    Attr find_attr(cstr name) {return iio_channel_find_attr(p, name);}
+    void enable(iio_channels_mask * mask) {iio_channel_enable(p, mask);}
+    void disable(iio_channels_mask * mask) {iio_channel_disable(p, mask);}
+    bool is_enabled(iio_channels_mask * mask) const { return iio_channel_is_enabled(p, mask);}
+    size_t read(Block block, void * dst, size_t len, bool raw) const; // Flawfinder: ignore
+    size_t write(Block block, void const * src, size_t len, bool raw);
     void set_data(void * data){iio_channel_set_data(p, data);}
-    size_t write(Buffer buffer, void const * src, size_t len);
-    size_t write_raw(Buffer buffer, void const * src, size_t len);
-
+    void * data() const {return iio_channel_get_data(p);}
+    iio_chan_type type() const {return iio_channel_get_type(p);}
+    iio_modifier modifier() const {return iio_channel_get_modifier(p);}
+    hwmon_chan_type hwmon_type() const {return hwmon_channel_get_type(p);}
+    unsigned long index() const { return impl::check_n(iio_channel_get_index(p), "iio_channel_get_index");}
+    iio_data_format const * data_format() const {return iio_channel_get_data_format(p);}
     void convert(void * dst, void const * src) const {iio_channel_convert(p, dst, src);}
     void convert_inverse(void * dst, void const * src) const {iio_channel_convert_inverse(p, dst, src);}
-    iio_data_format const * data_format() const {return iio_channel_get_data_format(p);}
-    unsigned long index() const { return impl::check_n(iio_channel_get_index(p), "iio_channel_get_index");}
 };
 
+/** @brief C++ wrapper for the @ref Stream C-API
+ */
+class Stream
+{
+    iio_stream * const p;
+public:
+
+    Stream() = delete;
+    Stream(iio_stream * s) : p(s){}
+    operator iio_stream * () const {return p;}
+
+    Block next_block() {return const_cast<iio_block *>(impl::check(iio_stream_get_next_block(p), "iio_stream_get_next_block")); }
```

Comment:
Won't the iio_block be deleted when it gets out of scope?

Reply:
No, the Block-class does not manage the lifetime of the referenced object. It is just a shallow wrapper for calling its functions.
See the comment at the top of the header. This should explain how the C++ classes manage resources.

Reply:
Ok, thanks.